compiler/lib/src/main/scala/codegen/CppWriter/ComponentCppWriter/ComponentCppWriter.scala

```diff
@@ -107,9 +107,9 @@ case class ComponentCppWriter (
       guardedList (hasEvents) (List("Fw/Log/LogString.hpp"))
     val internalStrHeaders =
       guardedList (hasInternalPorts) (List("Fw/Types/InternalInterfaceString.hpp"))
-
-    val systemHeaders = List(
-      "FpConfig.hpp"
+    val systemHeaders = List.concat(
+      List("FpConfig.hpp"),
+      guardedList (hasEvents) (List("atomic"))
```

Comment:
Simplify this

Reply:
Can we also put `atomic` first, to maintain alphabetical order? I think we can add `.sorted` at the end of the expression.

Reply:
Addressed in latest commit.